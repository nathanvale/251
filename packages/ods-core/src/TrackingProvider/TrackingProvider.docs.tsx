/* eslint-disable no-alert */
/* eslint-disable react/display-name */
import React from "react";
import { ComponentDocs, ExampleDocs } from "@origin-digital/ods-types";
import {
  TrackingProvider,
  TrackingProviderProps,
  Button,
  Card,
  Text,
  TextLink,
} from "..";

export const setupTracking: ExampleDocs = {
  label: "Setting up tracking in your app",
  description: (
    <Text>
      To provide tracking, all you have to do is wrap{" "}
      <TextLink href="#/components/TrackingProvider">TrackingProvider</TextLink>{" "}
      around the root of your app and pass in a handler that expects a tracking
      event object that contains a data-id, type of interaction component (e.g.
      Button), a label (e.g. Button label) and an optional postClickState (e.g.
      An accordion that has opened or closed).
    </Text>
  ),
  noCard: true,
  Code: () => (
    <TrackingProvider
      onTrackingCapture={(props) => {
        alert(
          `data-id: ${props["data-id"]}, type: ${props.type}, label: ${props.label}, postClickState: ${props.postClickState}`
        );
      }}
    >
      <Card>
        <Button>I am being tracked!</Button>
      </Card>
    </TrackingProvider>
  ),
};

export const setupReportingClient: ExampleDocs = {
  label: "Hooking in the @origin-digital/reporting-client",
  playroom: false,
  codeString: `import {
  interaction,
  IInteractionData
} from "@origin-digital/reporting-client";

const App = ()=><TrackingProvider
  onTrackingCapture={({ "data-id": dataId , label, type, postClickState }) => {
    const interactionData: IInteractionData = {
      dataId,
      componentType: type,
      label,
      state: postClickState
    };
    return interaction("click", interactionData);
  }}
>
  <Button>
    I am being tracked by the @origin-digital/reporting-client!
  </Button>
</TrackingProvider>`,
};

export const docs: ComponentDocs<TrackingProviderProps> = {
  category: "Logic",
  componentName: "TrackingProvider",
  description:
    "The TrackingProvider allows use to pass in a tracking handler that captures tracking events from ODS interaction components.",
  propDescriptions: {
    onTrackingCapture:
      "A tracking handler that that expects a TrackingEvent type.",
  },
  migrationGuide: false,
  examples: {
    default: {},
    additional: [setupTracking, setupReportingClient],
  },
  snippets: [],
};
