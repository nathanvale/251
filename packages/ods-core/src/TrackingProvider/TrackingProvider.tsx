import React, { createContext } from "react";
import { TrackingEventHandler } from "@origin-digital/ods-types";

export interface TrackingProviderProps {
  children: React.ReactNode;
  onTrackingCapture: TrackingEventHandler;
}

export const TrackingContext = createContext<
  | undefined
  | {
      onTrackingCapture: TrackingEventHandler;
    }
>(undefined);

export const TrackingProvider = ({
  children,
  onTrackingCapture,
}: TrackingProviderProps) => {
  return (
    <TrackingContext.Provider value={{ onTrackingCapture }}>
      {children}
    </TrackingContext.Provider>
  );
};

TrackingProvider.displayName = "TrackingProvider";
