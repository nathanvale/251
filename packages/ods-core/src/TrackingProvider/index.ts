export { TrackingProvider } from "./TrackingProvider";
export { useTracking } from "./useTracking";

export type TrackingProviderProps = import("./TrackingProvider").TrackingProviderProps;
