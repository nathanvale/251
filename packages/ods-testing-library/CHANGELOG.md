# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

# [0.3.0](https://bitbucket.orgn.io/od/origin-ui/compare/@origin-digital/ods-testing-library@0.2.0...@origin-digital/ods-testing-library@0.3.0) (2020-02-12)


### Features

* **tests, columns:** alignY default changed to top ([c23c964](https://bitbucket.orgn.io/od/origin-ui/commits/c23c964))





# 0.2.0 (2020-01-31)


### Features

* set up jest for snapshots ([6e6fbfa](https://bitbucket.orgn.io/od/origin-ui/commits/6e6fbfa))
