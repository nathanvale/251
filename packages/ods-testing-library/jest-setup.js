import { configure } from "@testing-library/react";

require("jest-styled-components");

configure({ testIdAttribute: "data-id" });
