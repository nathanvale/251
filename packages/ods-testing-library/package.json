{
  "name": "@origin-digital/ods-testing-library",
  "description": "ODS testing tools that enhance react-testing-library",
  "license": "UNLICENSED",
  "author": "Nathan Vale <nathan.vale@origin.com.au>",
  "version": "0.5.0",
  "main": "dist/cjs/index.js",
  "module": "dist/es/index.js",
  "files": [
    "dist",
    "jest-setup.js"
  ],
  "scripts": {
    "build": "yarn build:cjs && yarn build:es",
    "build:cjs": "BUILD_FORMAT=cjs origin-scripts build --monorepo --delete-dir-on-start --out-dir dist/cjs",
    "build:es": "BUILD_WEBPACK=true origin-scripts build --monorepo --out-dir dist/es"
  },
  "sideEffects": false,
  "typings": "dist/typings/index.d.ts",
  "dependencies": {
    "@babel/runtime-corejs3": "^7.8.3",
    "@testing-library/jest-dom": "^5.1.1"
  },
  "peerDependencies": {
    "@origin-digital/ods-themes": "*",
    "@testing-library/react": "9.x",
    "jest-styled-components": "6.x",
    "react": "^16.9.0"
  },
  "devDependencies": {
    "@testing-library/react": "9.x"
  },
  "publishConfig": {
    "registry": "https://artifactory.origindigital-dac.com.au/artifactory/api/npm/npm-internal/"
  }
}
