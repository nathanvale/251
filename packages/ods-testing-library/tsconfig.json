{
  "extends": "../../tsconfig.json",
  "compilerOptions": {
    "outDir": "./dist/typings",
    "rootDir": "./src"
  },
  "include": [
    "src",
    "src/**/*.json",
    "../../globals.d.ts",
    "../../@types",
    "jest-setup.ts"
  ],
  "references": [{"path": "../ods-themes/tsconfig.json"}]
}
