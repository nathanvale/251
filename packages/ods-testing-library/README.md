# @origin-digital/ods-testing-library

<!-- START doctoc generated TOC please keep comment here to allow auto update -->
<!-- DON'T EDIT THIS SECTION, INSTEAD RE-RUN doctoc TO UPDATE -->

- [Installation](#installation)
- [Why do I need this package?](#why-do-i-need-this-package)

<!-- END doctoc generated TOC please keep comment here to allow auto update -->

This package enhances the render function of `react-testing-library` so that the `<OriginThemeProvider/>` is wrapped around your rendered component.

All other API features of `react-testing-library` are also included.

## Installation

Install the package in your project directory with:

```sh
yarn add @origin-digital/ods-testing-library
```

This package also has a peer dependency on the following:

```sh
// with yarn
yarn add @origin-digital/ods-core
yarn add -D @testing-library/react
yarn add -D jest-styled-components
```

## Why do I need this package?

**The problem**

When using react-testing-library and ODS with the jest testing framework it is necessary to wrap **all** your components in our `<OriginThemeProvider/>` - otherwise the react component cannot be rendered and tested.

**The solution**

`@origin-digital/ods-testing-library` creates a wrapper around the `react-testing-library` render function. It is here that we wrap the `<OriginThemeProvider/>`. All other API features of `react-testing-library` are also included.

This practice is recommended by the `react-testing-library` documentation itself...

https://testing-library.com/docs/native-testing-library/setup#custom-render

The origin-ui dogfoods this package itself but it can also be used in your own app to test by adding to your jest config:

```
{
...,
setupFilesAfterEnv: [@origin-digital/ods-testing-library/jest-setup],
...
}
```

and usage in your tests:

```
import React from 'react';
import {render} from '@origin-digital/ods-testing-library';
import {Box} from '../Box/Box';
import {Stack} from './Stack';

test('It can stack', () => {
  const {container} = render(
    <Stack>
      <Box backgroundColor="red" padding="medium" />
      <Box backgroundColor="orange" padding="medium" />
      <Box backgroundColor="blue" padding="medium" />
    </Stack>,
  );
  expect(container.firstChild).toMatchSnapshot();
});
```
