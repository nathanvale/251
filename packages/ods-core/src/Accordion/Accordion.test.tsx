import React from "react";
import { render, fireEvent } from "@origin-digital/ods-testing-library";
import { TrackingEventHandler } from "@origin-digital/ods-types";
import { TrackingProvider } from "../TrackingProvider/TrackingProvider";
import {
  generateAccordion,
  generateControlledAccordion,
} from "./Accordion.helper";
import {
  AccordionBaseProps,
  AccordionControlledProps,
  AccordionUncontrolledProps,
} from "./Accordion";

const renderAccordion = (
  accordionProps: Partial<AccordionBaseProps> & AccordionUncontrolledProps = {}
) => render(generateAccordion(accordionProps));

const renderControlledAccordion = (
  accordionProps: Partial<AccordionBaseProps> & AccordionControlledProps
) => render(generateControlledAccordion(accordionProps));

test("It renders", () => {
  const { container } = renderAccordion();
  expect(container).toMatchSnapshot();
});

test("It uses small padding", () => {
  const { container } = renderAccordion({ size: "small" });
  expect(container).toMatchSnapshot();
});

test("It uses medium padding", () => {
  const { container } = renderAccordion({ size: "medium" });
  expect(container).toMatchSnapshot();
});

test("It sets the id's", () => {
  const { getByTestId } = renderAccordion({ id: "my-id" });
  expect(getByTestId("accordion-header")).toHaveAttribute("id", "my-id-header");
  expect(getByTestId("accordion-content")).toHaveAttribute(
    "id",
    "my-id-content"
  );
});

test("It sets the data-id", () => {
  const { getAllByTestId } = renderAccordion({ "data-id": "my-id" });
  expect(getAllByTestId("my-id")).toHaveLength(1);
});

test("It adds additional class", () => {
  const { getByTestId } = renderAccordion({ className: "my-class" });
  expect(getByTestId("accordion")).toHaveClass("my-class");
});

test("It calls onChange handler", () => {
  const onChange = jest.fn();

  const { getByTestId } = renderControlledAccordion({
    expanded: false,
    onChange,
  });
  fireEvent.click(getByTestId("accordion-header"));
  expect(onChange).toHaveBeenCalled();
});

it("should call tracking handler when passed into the TrackingProvider", () => {
  const summary = "My accordion";

  const onTrackingCapture: TrackingEventHandler = jest.fn();

  const { getByTestId } = render(
    <TrackingProvider onTrackingCapture={onTrackingCapture}>
      {generateAccordion({ summary })}
    </TrackingProvider>
  );

  const expectedValues = {
    "data-id": "accordion",
    type: "Accordion",
    label: summary,
  };

  fireEvent.click(getByTestId("accordion-header"));
  expect(onTrackingCapture).toHaveBeenCalledWith({
    ...expectedValues,
    postClickState: "expanded",
  });

  fireEvent.click(getByTestId("accordion-header"));
  expect(onTrackingCapture).toHaveBeenCalledWith({
    ...expectedValues,
    postClickState: "collapsed",
  });
});
