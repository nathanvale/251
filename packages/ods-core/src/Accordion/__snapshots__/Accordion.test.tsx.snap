// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`It renders 1`] = `
.c2 {
  margin: 0;
  padding: 0;
  border: 0;
  box-sizing: border-box;
  font-size: 100%;
  font: inherit;
  vertical-align: baseline;
  text-align: left;
}

.c1 {
  margin: 0;
  padding: 0;
  border: 0;
  box-sizing: border-box;
  font-size: 100%;
  font: inherit;
  vertical-align: baseline;
  text-align: left;
}

.c0 {
  max-width: 100%;
}

<div>
  <div
    class="MuiPaper-root MuiAccordion-root Accordion-accordion-1 Accordion-accordion-4 MuiAccordion-rounded MuiPaper-elevation1 MuiPaper-rounded"
    data-id="accordion"
  >
    <div
      aria-controls="my-accordion-content"
      aria-disabled="false"
      aria-expanded="false"
      class="MuiButtonBase-root MuiAccordionSummary-root Accordion-accordionSummary-2 Accordion-accordionSummary-5"
      data-id="accordion-header"
      id="my-accordion-header"
      role="button"
      tabindex="0"
    >
      <div
        class="MuiAccordionSummary-content"
      >
        <h4
          class="typography-body-variant-basekick-7 typography-neutral-color-8 typography-medium-weight-9 typography-strong-support-10 c0 -h4 c1"
          data-id="text"
        >
          My accordion title
        </h4>
      </div>
      <div
        aria-disabled="false"
        aria-hidden="true"
        class="MuiButtonBase-root MuiIconButton-root MuiAccordionSummary-expandIcon MuiIconButton-edgeEnd"
      >
        <span
          class="MuiIconButton-label"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root svg-icon-critical-small-svg-icon-12"
            data-id="IconExpand"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
            />
          </svg>
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </div>
    </div>
    <div
      class="MuiCollapse-container MuiCollapse-hidden"
      style="min-height: 0px;"
    >
      <div
        class="MuiCollapse-wrapper"
      >
        <div
          class="MuiCollapse-wrapperInner"
        >
          <div
            aria-labelledby="my-accordion-header"
            id="my-accordion-content"
            role="region"
          >
            <div
              class="MuiAccordionDetails-root Accordion-accordionDetails-3 Accordion-accordionDetails-6"
              data-id="accordion-content"
              id="my-accordion-content"
            >
              <span
                class="typography-body-variant-basekick-13 typography-neutral-color-14 typography-regular-weight-15 typography-strong-support-16 c0 -span c2"
                data-id="text"
              >
                Lorem ipsum dolor sit amet, consectetur adipiscing elit. Suspendisse facilisis posuere sodales. Aliquam sit amet mauris ut metus fringilla ullamcorper. Suspendisse maximus mi at fringilla convallis. Sed pretium vitae libero sed vestibulum. Quisque suscipit vitae ipsum vel sollicitudin. In mollis quam neque.
              </span>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`It uses medium padding 1`] = `
.c2 {
  margin: 0;
  padding: 0;
  border: 0;
  box-sizing: border-box;
  font-size: 100%;
  font: inherit;
  vertical-align: baseline;
  text-align: left;
}

.c1 {
  margin: 0;
  padding: 0;
  border: 0;
  box-sizing: border-box;
  font-size: 100%;
  font: inherit;
  vertical-align: baseline;
  text-align: left;
}

.c0 {
  max-width: 100%;
}

<div>
  <div
    class="MuiPaper-root MuiAccordion-root Accordion-accordion-35 Accordion-accordion-38 MuiAccordion-rounded MuiPaper-elevation1 MuiPaper-rounded"
    data-id="accordion"
  >
    <div
      aria-controls="my-accordion-content"
      aria-disabled="false"
      aria-expanded="false"
      class="MuiButtonBase-root MuiAccordionSummary-root Accordion-accordionSummary-36 Accordion-accordionSummary-39"
      data-id="accordion-header"
      id="my-accordion-header"
      role="button"
      tabindex="0"
    >
      <div
        class="MuiAccordionSummary-content"
      >
        <h4
          class="typography-body-variant-basekick-41 typography-neutral-color-42 typography-medium-weight-43 typography-strong-support-44 c0 -h4 c1"
          data-id="text"
        >
          My accordion title
        </h4>
      </div>
      <div
        aria-disabled="false"
        aria-hidden="true"
        class="MuiButtonBase-root MuiIconButton-root MuiAccordionSummary-expandIcon MuiIconButton-edgeEnd"
      >
        <span
          class="MuiIconButton-label"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root svg-icon-critical-small-svg-icon-46"
            data-id="IconExpand"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
            />
          </svg>
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </div>
    </div>
    <div
      class="MuiCollapse-container MuiCollapse-hidden"
      style="min-height: 0px;"
    >
      <div
        class="MuiCollapse-wrapper"
      >
        <div
          class="MuiCollapse-wrapperInner"
        >
          <div
            aria-labelledby="my-accordion-header"
            id="my-accordion-content"
            role="region"
          >
            <div
              class="MuiAccordionDetails-root Accordion-accordionDetails-37 Accordion-accordionDetails-40"
              data-id="accordion-content"
              id="my-accordion-content"
            >
              <span
                class="typography-body-variant-basekick-47 typography-neutral-color-48 typography-regular-weight-49 typography-strong-support-50 c0 -span c2"
                data-id="text"
              >
                Lorem ipsum dolor sit amet, consectetur adipiscing elit. Suspendisse facilisis posuere sodales. Aliquam sit amet mauris ut metus fringilla ullamcorper. Suspendisse maximus mi at fringilla convallis. Sed pretium vitae libero sed vestibulum. Quisque suscipit vitae ipsum vel sollicitudin. In mollis quam neque.
              </span>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`It uses small padding 1`] = `
.c2 {
  margin: 0;
  padding: 0;
  border: 0;
  box-sizing: border-box;
  font-size: 100%;
  font: inherit;
  vertical-align: baseline;
  text-align: left;
}

.c1 {
  margin: 0;
  padding: 0;
  border: 0;
  box-sizing: border-box;
  font-size: 100%;
  font: inherit;
  vertical-align: baseline;
  text-align: left;
}

.c0 {
  max-width: 100%;
}

<div>
  <div
    class="MuiPaper-root MuiAccordion-root Accordion-accordion-18 Accordion-accordion-21 MuiAccordion-rounded MuiPaper-elevation1 MuiPaper-rounded"
    data-id="accordion"
  >
    <div
      aria-controls="my-accordion-content"
      aria-disabled="false"
      aria-expanded="false"
      class="MuiButtonBase-root MuiAccordionSummary-root Accordion-accordionSummary-19 Accordion-accordionSummary-22"
      data-id="accordion-header"
      id="my-accordion-header"
      role="button"
      tabindex="0"
    >
      <div
        class="MuiAccordionSummary-content"
      >
        <h4
          class="typography-body-variant-basekick-24 typography-neutral-color-25 typography-medium-weight-26 typography-strong-support-27 c0 -h4 c1"
          data-id="text"
        >
          My accordion title
        </h4>
      </div>
      <div
        aria-disabled="false"
        aria-hidden="true"
        class="MuiButtonBase-root MuiIconButton-root MuiAccordionSummary-expandIcon MuiIconButton-edgeEnd"
      >
        <span
          class="MuiIconButton-label"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root svg-icon-critical-small-svg-icon-29"
            data-id="IconExpand"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
            />
          </svg>
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </div>
    </div>
    <div
      class="MuiCollapse-container MuiCollapse-hidden"
      style="min-height: 0px;"
    >
      <div
        class="MuiCollapse-wrapper"
      >
        <div
          class="MuiCollapse-wrapperInner"
        >
          <div
            aria-labelledby="my-accordion-header"
            id="my-accordion-content"
            role="region"
          >
            <div
              class="MuiAccordionDetails-root Accordion-accordionDetails-20 Accordion-accordionDetails-23"
              data-id="accordion-content"
              id="my-accordion-content"
            >
              <span
                class="typography-body-variant-basekick-30 typography-neutral-color-31 typography-regular-weight-32 typography-strong-support-33 c0 -span c2"
                data-id="text"
              >
                Lorem ipsum dolor sit amet, consectetur adipiscing elit. Suspendisse facilisis posuere sodales. Aliquam sit amet mauris ut metus fringilla ullamcorper. Suspendisse maximus mi at fringilla convallis. Sed pretium vitae libero sed vestibulum. Quisque suscipit vitae ipsum vel sollicitudin. In mollis quam neque.
              </span>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
